schema/helpers/base/parameters.go

```diff
@@ -52,8 +53,9 @@ func (parameters parameters) Equal(Parameters helpers.Parameters) bool {
 }
 
 func (parameters parameters) Fetch(context sdkTypes.Context, id types.ID) (parameter types.Parameter) {
-	parameters.paramsSubspace.Get(context, id.Bytes(), &parameter)
-	return parameter
+	var data types.Data
+	parameters.paramsSubspace.Get(context, id.Bytes(), &data)
+	return base.NewParameter(id, data, func(interface{}) error { return nil })
```

Comment:
need a better way to do this, possibly find a way to add the validatorFunc.
it should either be passed into the function, or the parameters should have a prototype of some sorts present.

Reply:
parameters Prototype() method returns a default list of parameter[] populated with validate methods. When fetching, the values are fetched. Can then do a get of parameter with same id on parameters, which will have the default value and validate method. Can then replace default value with fetched value.

Reply:
there is no prototype method, but the parameters that are being used for fetching in genesis.go, are params of genesis, which have the list. so directly using those.